src/client/runner/runner.js

```diff
@@ -167,7 +161,7 @@ Runner.prototype._onTestError = function (err, isAssertion) {
     return errorProcessingChain;
 };
 
-Runner.prototype._onFatalError = function (err) {
+Runner.prototype._onFatalError = function (err, isSavedError) {
```

Comment:
I don't like the `isSavedError` argument because I can't get what does it mean without reading long notes and understanding `beforeunload event handler` logic.
Also we have to duplicate stepName escaping with this approach:
https://github.com/DevExpress/testcafe/pull/330/files#diff-619aff4845280917d61f0dfe5e1a6071R179
https://github.com/DevExpress/testcafe/pull/330/files#diff-619aff4845280917d61f0dfe5e1a6071R149

I suppose we should call `this._reportErrorToServer` from `this._onTestError` only. So we can do the following:
```js
//Runner.prototype._onFatalError = function (err) {
if (beforeUnloadRaised) {
        err.skipScreenshot = true;
        this.testContextStorage.setProperty('testError', err);
} else {
    this
        ._onTestError(err)
        .then(Runner.checkStatus);
}
```

```js
//Runner.prototype._onTestError = function (err, isAssertion) {
err.screenshotRequired = !err.skipScreenshot && SETTINGS.get()....
```